lib/spack/docs/getting_started.rst

```diff
@@ -57,3 +112,928 @@ This will install a new spack script in ``/my/favorite/prefix/bin``,
+With Spack, things usually "just work" with GCC.  Not so for other
+compilers.  This section provides details on how to get specific
+compilers working.
+
+^^^^^^^^^^^^^^^
+Intel Compilers
+^^^^^^^^^^^^^^^
+
+Intel compilers are unusual because a single Intel compiler version
+can emulate multiple GCC versions.  In order to provide this
+functionality, the Intel compiler needs GCC to be installed.
+Therefore, the following steps are necessary to successfully use Intel
+compilers:
+
+ #. Install a version of GCC that implements the desired language
+    features (``spack install gcc``).
+
+ #. Tell the Intel compiler how to find that desired GCC.  This may be
+    done in one of two ways: (text taken from `Intel Reference Guide
+    <https://software.intel.com/en-us/node/522750>`_):
+
+    > By default, the compiler determines which version of ``gcc`` or ``g++``
+    > you have installed from the ``PATH`` environment variable.
+    > 
+    > If you want use a version of ``gcc`` or ``g++`` other than the default
+    > version on your system, you need to use either the ``-gcc-name``
+    > or ``-gxx-name`` compiler option to specify the path to the version of
+    > ``gcc`` or ``g++`` that you want to use.
+
+Intel compilers may therefore be configured in one of two ways with
+Spack: using modules, or using compiler flags.
+
+""""""""""""""""""""""""""
+Configuration with Modules
+""""""""""""""""""""""""""
+
+One can control which GCC is seen by the Intel compiler with modules.
+A module must be loaded both for the Intel Compiler (so it will run)
+and GCC (so the compiler can find the intended GCC).  The following
+configuration in ``compilers.yaml`` illustrates this technique:
+
+    .. code-block:: yaml
+
+        compilers:
+        - compiler:
+            modules = [gcc-4.9.3, intel-15.0.24]
+            operating_system: centos7
+            paths:
+              cc: /opt/intel-15.0.24/bin/icc-15.0.24-beta
+              cxx: /opt/intel-15.0.24/bin/icpc-15.0.24-beta
+              f77: /opt/intel-15.0.24/bin/ifort-15.0.24-beta
+              fc: /opt/intel-15.0.24/bin/ifort-15.0.24-beta
+            spec: intel@15.0.24.4.9.3
+
+
+    .. note::
+
+        The version number on the Intel compiler is a combination of
+        the "native" Intel version number and the GNU compiler it is
+        targeting.
+
+    .. warning::
+
+        This solution has not yet been tested.  Details may vary.
+
+""""""""""""""""""""""""""
+Command Line Configuration
+""""""""""""""""""""""""""
+
+. warning::
+
+    As of the writing of this manual, added compilers flags are broken;
+    see `GitHub Issue <https://github.com/LLNL/spack/pull/1532>`_.
+
+One can also control which GCC is seen by the Intel compiler by adding
+flags to the ``icc`` command:
+
+ #. Identify the location of the compiler you just installed:
+
+    .. code-block:: console
+
+        $ spack location -i gcc
+        /home2/rpfische/spack2/opt/spack/linux-centos7-x86_64/gcc-4.9.3-iy4rw...
```

Comment:
It looks like `spack location -i` just gives the location, `spack find -p`
formats it a bit more for being human-readable.

On Wed, Sep 7, 2016 at 4:28 PM, Adam J. Stewart <notifications@github.com>
wrote:

> In lib/spack/docs/getting_started.rst:
>
> > +""""""""""""""""""""""""""
> > +
> > +. warning::
> > +
> > +    As of the writing of this manual, added compilers flags are broken;
> > +    see `GitHub Issue <https://github.com/LLNL/spack/pull/1532>`_.
> > +
> > +One can also control which GCC is seen by the Intel compiler by adding
> > +flags to the ``icc`` command:
> > +
> > + #. Identify the location of the compiler you just installed:
> > +
> > +    .. code-block:: console
> > +
> > +        $ spack location -i gcc
> > +        /home2/rpfische/spack2/opt/spack/linux-centos7-x86_64/gcc-4.9.3-iy4rw...
>
> Is spack location -i identical to spack find -p?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1676/files/c9e83c62a002c42852fcb54a3b45d54a68cb8ccc#r77897280>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cd_mcE2o_bBnXf48isZbr2dxRu6vJks5qnx5vgaJpZM4JxOaa>
> .
>